src/containers/app/AppSagas.js

```diff
@@ -2,34 +2,159 @@
+import { submitDataGraphWorker } from '../../core/data/DataSagas';
 import { getEntityDataModelTypes } from '../../core/edm/EDMActions';
 import { getEntityDataModelTypesWorker } from '../../core/edm/EDMSagas';
+import { APP_TYPE_FQNS, PROPERTY_TYPE_FQNS } from '../../core/edm/constants/FullyQualifiedNames';
+import { ERR_ORGS_NOT_FOUND } from '../../utils/Errors';
+import { APP_NAME } from '../../utils/constants/GeneralConstants';
 
 const { getApp, getAppConfigs } = AppApiActions;
 const { getAppWorker, getAppConfigsWorker } = AppApiSagas;
+const { getEntitySetData } = DataApiActions;
+const { getEntitySetDataWorker } = DataApiSagas;
+const { getOrganization, getUsersWithRole } = OrganizationsApiActions;
+const { getOrganizationWorker, getUsersWithRoleWorker } = OrganizationsApiSagas;
+const { getPropertyValue } = DataUtils;
+const {
+  COUNTY_ID,
+  ENTITY_KEY_ID,
+  FIRST_NAME,
+  LAST_NAME,
+} = PROPERTY_TYPE_FQNS;
 
 const LOG = new Logger('AppSagas');
 
+const personGivenNamePTID = 'e9a0b4dc-5298-47c1-8837-20af172379a5';
+const personSurnamePTID = '7b038634-a0b4-4ce1-a04f-85d1775937aa';
+const idPTID = '5260cfbd-bfa4-40c1-ade5-cd83cc9f99b2';
+
+/*
+ *
+ * AppActions.getCurrentStaff()
+ *
+ */
+
+function* getCurrentStaffWorker(action :SequenceAction) :Generator<*, *, *> {
+
+  try {
+    yield put(getCurrentStaff.request(action.id));
+    const { selectedAppConfig, staffESID } = action.value;
+
+    let response = yield call(getEntitySetDataWorker, getEntitySetData({ entitySetId: staffESID }));
+    if (response.error) throw response.error;
+    let staff :List = fromJS(response.data);
+
+    const orgId = selectedAppConfig.organization?.id;
+    response = yield call(getOrganizationWorker, getOrganization(selectedAppConfig.organization?.id));
+    if (response.error) throw response.error;
+
+    const readRole :?Object = response.data?.roles?.find((role) => role.title?.includes('READ'));
+    const readRoleId :?UUID = readRole?.id;
+
+    response = yield call(getUsersWithRoleWorker, getUsersWithRole({ organizationId: orgId, roleId: readRoleId }));
+    if (response.error) throw response.error;
+
+    const authorizedUsers = Set().withMutations((mutator) => {
+      response.data.forEach((user) => {
+        mutator.add(user);
+      });
+    });
+    const authorizedUserEmails = authorizedUsers.map((user) => user.email);
+
+    const entityData = {
+      [staffESID]: []
+    };
+
+    staff.forEach((staffMember) => {
+      const userEmail = getPropertyValue(staffMember, [COUNTY_ID, 0]);
+      if (!authorizedUserEmails.includes(userEmail)) {
+        const user = authorizedUsers.find((authorizedUser) => authorizedUser.email === userEmail);
+        const userFirstName = user.given_name;
+        const userLastName = user.family_name;
+        const { email } = user;
+        entityData[staffESID].push({
+          [personGivenNamePTID]: [userFirstName],
+          [personSurnamePTID]: [userLastName],
+          [idPTID]: email,
```

Comment:
just curious why property multiplicity on the firstname & lastname but not email.

Reply:
oops this is a mistake, thanks for catching it